workspaces/ui-v2/src/pages/docs/components/RenderBody.tsx

```diff
@@ -53,7 +60,12 @@ export function TwoColumnBodyEditable(props: TwoColumnBodyProps) {
       }
     />
   );
-}
+};
+
+// Memoize this as there could be a large number of contribution groups
```

Comment:
It's probably also worth virtualizating this since huge specs cause a lot of rerenders outside of the viewport, leading to slow responsiveness. 

The good thing about redux here though is that we have more fine grained control over renders making performance improvements (where we need it) much simpler

Reply:
Possible premature optimisation, or did we already have concrete cases of this being a performance bottleneck? Memoization has a cost and can actually lead to degraded performance, especially when applied in multiple places in the hierarchy, which can be tricky to reason about. Lets not apply any of it until there's concrete evidence that's where lots unnecessary renders happen, and then memoisation reduces those numbers.

Reply:
This does actually cause an issue on the `debug-github-capture` example - there's ~200 fields to render, and on first render (starting edit flow) - there's 200 fields to update at once (which needs to render the input field).

Reply:
This fixes the case where we _initially_ type into a new field (which changes pendingCount -> causes a rerender int he `EndpointRootPage` component -> causing this to rerender when it doesn't need to) - it does not fix the case where we actually need to rerender these fields (where if we have ~200 fields, we have to render a large number of UI elements which causes UI jank - in which case virtualizing this could help)